src/app/asset-page/asset-page.component.ts

```diff
@@ -782,23 +785,25 @@ export class AssetPage implements OnInit, OnDestroy {
         .subscribe((blob) => {
           if (blob.size > 0) {
               if (this.isMSAgent) {
-                  this.downloadLoading = false
-                  this.navigator.msSaveBlob(blob, 'download.jpg')
+                  // We give MS Browsers an extra pause of 2 secs
+                  setTimeout(() => {
+                    this.downloadViewReady = true
```

Comment:
Looks like this variable is set multiple times - are there any cases when this might be setting and it's not supposed to?

Reply:
In runDownloadView - it is set to false - then re-enabled just before the file save dialog is triggered.